src/torchmetrics/detection/mean_ap.py

```diff
@@ -475,11 +478,11 @@ def _compute_iou(self, idx: int, class_id: int, max_det: int) -> Tensor:
         """
 
         # if self.iou_type == "bbox":
-        gt = self.groundtruths[idx]
-        det = self.detections[idx]
+        gt = self.groundtruths[idx].to(self.device)
```

Comment:
these are tuples so you can't move them simply to the device, you need to do it per item or can we keep them in CPU?

Reply:
They are not tuples, the are list of tensors. So, by accessing an element via the index, we get a Tensor that can be moved to the device.

And I can see it through a run:

```
type(self.groundtruths[idx])
<class 'torch.Tensor'>
```

And the list is declared on line 360:

```
groundtruths: List[Tensor]
```

And the same for the other one:

```
type(self.detections[idx])
<class 'torch.Tensor'>
```

@Borda

Reply:
ok, so there are some data types which are tuple as you can see from the failing tests :)

Reply:
Ah. Those tests were passing before I rebased with the Master branch. But let me check them and understand what's going on.

Reply:
Hey @Borda ,

All failing jobs relate to this:

```
INFO:root:requirements_path='requirements.txt' with torch_version='None'
INFO:root:finding ecosystem versions for: 1.13.1
Traceback (most recent call last):
  File "./requirements/adjust-versions.py", line 83, in <module>
    requirements = adjust(requirements, torch_version)
  File "./requirements/adjust-versions.py", line 64, in adjust
    latest = find_latest(torch_version)
  File "./requirements/adjust-versions.py", line 51, in find_latest
    raise ValueError(f"Missing {ver} in {VERSIONS}")
ValueError: Missing 1.13.1 in [{'torch': '1.13.0', 'torchvision': '0.14.0', 'torchtext': '0.14.0'}, {'torch': '1.12.1', 'torchvision': '0.13.1', 'torchtext': '0.13.1'}, {'torch': '1.12.0', 'torchvision': '0.13.0', 'torchtext': '0.13.0'}, {'torch': '1.11.0', 'torchvision': '0.12.0', 'torchtext': '0.12.0'}, {'torch': '1.10.2', 'torchvision': '0.11.3', 'torchtext': '0.11.2'}, {'torch': '1.10.1', 'torchvision': '0.11.2', 'torchtext': '0.11.1'}, {'torch': '1.10.0', 'torchvision': '0.11.1', 'torchtext': '0.11.0'}, {'torch': '1.9.1', 'torchvision': '0.10.1', 'torchtext': '0.10.1'}, {'torch': '1.9.0', 'torchvision': '0.10.0', 'torchtext': '0.10.0'}, {'torch': '1.8.2', 'torchvision': '0.9.1', 'torchtext': '0.9.1'}, {'torch': '1.8.1', 'torchvision': '0.9.1', 'torchtext': '0.9.1'}, {'torch': '1.8.0', 'torchvision': '0.9.0', 'torchtext': '0.9.0'}]
```

Any idea what's going on? I mean, it's clear is version related, but why it changed after the rebase?

Reply:
@wilderrodrigues this is just bad luck as torch 1.13.1 was released between the test runs.

Reply:
> All failing jobs relate to this:

it was fixed on master